packages/fdc3-agent-proxy/src/util.ts

```diff
@@ -0,0 +1,20 @@
+import { AgentEventMessage, AgentResponseMessage } from '@kite9/fdc3-schema/generated/api/BrowserTypes';
+import { ChannelError, OpenError, ResolveError } from '@kite9/fdc3-standard';
+
+export type ErrorMessages = ChannelError | OpenError | ResolveError;
+
+/** Utility function that logs and throws a specified error if a specified property does not exist.
+ *  Used to lightly validate messages being processed primarily to catch errors in Desktop Agent
+ *  implementations.
+ */
+export const throwIfUndefined = (
+  property: object | string | number | null | undefined,
+  absentMessage: string,
+  message: AgentResponseMessage | AgentEventMessage,
+  absentError: ErrorMessages
+): void => {
+  if (property === undefined) {
```

Comment:
can we add a util test for this, like you did with logging?

Reply:
Sure - we should perhaps do the same for AbstractMessaging's `exchange` and `waitFor` to make sure they continue to throw on errors properly.